### PR TITLE
Nü Note Options: Mark V

### PR DESCRIPTION
**Pull request by jleandroperez**
### 🛑  Important
Don't merge until #906 #907 #909 and #910 are ready: everything must be merged in a single shot!

### Details:
- **Wires Collaboration support** in the Note Options Interface
- **OptionsViewController** now implements the AddCollaborator delegate protocol
- **Editor:** Drops the **blankNote** flag
- **Note:** New **isBlank** and **emailTags** properties

@frosty Sir!! Can I bug you with this one?
Thanks in advance!!!

Ref. #763

### Test: Collaboration
1. Open any note
2. Press over the top `( i )` icon
3. Press over **Collaborate**

- [ ] Verify you can add / remove collaborators
- [ ] Verify that newly added collaborators get access to the shared document

### Test: Empty Notes
1. Launch Simplenote and press the top right `+` button
2. **Do not type anything**
3. Press back

- [ ] Verify that the newly added note is deleted

### Test: Non Empty Notes
1. Launch Simplenote and press the top right `+` button
2. Type anything
3. Delete the contents
4. Press back

- [ ] Verify that the newly added is **not** deleted

### Release
`RELEASE-NOTES.txt` was updated in cdedacd with:
 
> New Note Options interface #763

### CODE REVIEW

File: Simplenote/Classes/Note+Properties.swift [hunk 1]
```diff
@@ -5,6 +5,13 @@ import Foundation
 //
 extension Note {
 
+    /// Indicates if the receiver is a blank document
+    ///
+    @objc
+    var isBlank: Bool {
+        objectID.isTemporaryID && content?.count == .zero && tagsArray?.count == .zero
```
Comment (frosty) [inline Simplenote/Classes/Note+Properties.swift:12]: A note can only be blank if it hasn't been saved?

File: Simplenote/Classes/Note.h [hunk 2]
```diff
@@ -34,20 +33,19 @@
...
+@property (nonatomic, strong) NSMutableArray<NSString*> *tagsArray;
+@property (nonatomic,   copy) NSString                  *shareURL;
+@property (nonatomic,   copy) NSDate                    *creationDate;
+@property (nonatomic,   copy) NSString                  *systemTags;
+@property (nonatomic,   copy) NSString                  *modificationDatePreview;
+@property (nonatomic,   copy) NSString                  *creationDatePreview;
 
-@property (nonatomic,   copy) NSString          *titlePreview;
-@property (nonatomic,   copy) NSString          *bodyPreview;
+@property (nonatomic,   copy) NSString                  *titlePreview;
```
Comment (frosty) [inline Simplenote/Classes/Note.h:47]: :chef-kiss:

**Review by frosty — APPROVED** [review]
**Test: Collaboration**

- [x] Verify you can add / remove collaborators
- [x] Verify that newly added collaborators get access to the shared document

**Test: Empty Notes**

- [x] Verify that the newly added note is deleted

**Test: Non Empty Notes**

- [x] Verify that the newly added is not deleted

Looks great! One small comment, but I think I understand the rationale now I've run through the testing steps – isBlank is to determine if it's a new, blank document? (and therefore delete it)

File: Simplenote/Classes/Note+Properties.swift (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/Classes/Note+Properties.swift:12]: Yes sir, that would be accurate!. Perhaps we should consider renaming `isBlank` to something more clear!

If it was never saved (and there is no content), it's safe to delete. The second you save it, `isTemporaryID` returns false, so it doesn't matter if it has (or doesn't have) any contents.

**Comment by jleandroperez** [comment]
@frosty **Thank you!!!!**

**Pull request merged**